core/src/main/java/org/elasticsearch/action/DocWriteResponse.java

```diff
@@ -242,7 +242,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (forcedRefresh) {
             builder.field("forced_refresh", forcedRefresh);
         }
-        shardInfo.toXContent(builder, params);
+        builder.field("_shards", shardInfo);
```

Comment:
I like when toXContent() starts it's own anonymous object because it makes it easy to include the object anywhere: array, field, as root object etc. But I understand your concern and in this case I agree it makes sense for ShardInfo to starts its own _shards object. So I reverted the change.